kDrive/UI/Controller/Files/Search/SearchFilesViewModel.swift

```diff
@@ -0,0 +1,246 @@
+    }
+
+    required init(driveFileManager: DriveFileManager, currentDirectory: File?) {
+        let configuration = Configuration(normalFolderHierarchy: false,
+                                          showUploadingFiles: false,
+                                          isMultipleSelectionEnabled: false,
+                                          rootTitle: KDriveResourcesStrings.Localizable.searchTitle,
+                                          emptyViewType: .noSearchResults,
+                                          leftBarButtons: [.cancel],
+                                          rightBarButtons: [.searchFilters],
+                                          sortingOptions: [.newer, .older, .relevance],
+                                          matomoViewPath: [MatomoUtils.Views.search.displayName])
+        filters = Filters()
+        let searchFakeRoot = driveFileManager.getManagedFile(from: DriveFileManager.searchFilesRootFile)
+        super.init(configuration: configuration, driveFileManager: driveFileManager, currentDirectory: searchFakeRoot)
+        files = AnyRealmCollection(AnyRealmCollection(searchFakeRoot.children).sorted(by: [sortType.value.sortDescriptor]))
+    }
+
+    override func startObservation() {
+        super.startObservation()
+        // Overriding default behavior to change list style in recent searches
+        listStyleObservation?.cancel()
+        listStyleObservation = nil
+        listStyle = .list
+        // Custom sort type
+        sortTypeObservation?.cancel()
+        sortTypeObservation = nil
+        sortType = .newer
+        sortingChanged()
+    }
+
+    override func loadFiles(cursor: String? = nil, forceRefresh: Bool = false) async throws {
+        guard isDisplayingSearchResults else { return }
+
+        var nextCursor: String?
+        if ReachabilityListener.instance.currentStatus == .offline {
+            searchOffline()
+        } else {
+            startRefreshing(cursor: cursor)
+
+            do {
+                (_, nextCursor) = try await driveFileManager.searchFile(query: currentSearchText,
+                                                                        date: filters.date?.dateInterval,
+                                                                        fileType: filters.fileType,
+                                                                        categories: Array(filters.categories),
+                                                                        belongToAllCategories: filters.belongToAllCategories,
+                                                                        cursor: cursor,
+                                                                        sortType: sortType)
+                endRefreshing()
+
+                if let nextCursor {
+                    try await loadFiles(cursor: nextCursor)
+                }
+            } catch let error as DriveError where error == .networkError {
+                searchOffline()
```

Comment:
`.searchOffline()` will run on the MainActor (I checked), can we mark the DriveFileManager.searchOffline async ?

Reply:
Seems risky, if we mark `searchOffline` async it could switch to an other thread and the resulting realm/objects (which are not frozen) won't be on the correct thread

Reply:
Sure, not a quickwin then. 
Still, I think it is a good thing to push to not do realm stuff in the main thread.